utilities/pr_commit_counts/pr_commit_counts.py

```diff
@@ -0,0 +1,72 @@
+import requests
+import os
+
+def get_paginated_results(url, headers):
+    """ Function to handle pagination for GitHub API requests """
+    results = []
+    while url:
+        response = requests.get(url, headers=headers)
+        if response.status_code == 200:
+            results.extend(response.json())
+            if 'next' in response.links.keys():
+                url = response.links['next']['url']
+            else:
+                url = None
+        else:
+            raise Exception(f"GitHub API request failed with status code {response.status_code}")
+    return results
+
+def get_repos(organization, headers):
+    """ Get a list of repositories for a given organization """
+    url = f"https://api.github.com/orgs/{organization}/repos"
+    return get_paginated_results(url, headers)
+
+def get_pull_requests(organization, repo, headers):
+    """ Get a list of pull requests for a given repository """
+    url = f"https://api.github.com/repos/{organization}/{repo}/pulls?state=all"
+    return get_paginated_results(url, headers)
+
+def get_commits_for_pull_request(organization, repo, pull_number, headers):
+    """ Get a list of commits for a given pull request """
+    url = f"https://api.github.com/repos/{organization}/{repo}/pulls/{pull_number}/commits"
+    return get_paginated_results(url, headers)
+
+def main():
+    # organization = input("Enter the organization name: ")
+    # github_token = input("Enter your GitHub access token: ")
+
+    organization = "ORGANIZATION_NAME" # Replace with your organization name
+    github_token = os.getenv('SEMGREP_API_WEB_TOKEN') # Add your GH token as a environment variable
+
+    headers = {
+        'Authorization': f'token {github_token}',
+        'Accept': 'application/vnd.github.v3+json'
+    }
+
+    total_commits = 0
+    total_prs = 0
+
+    repos = get_repos(organization, headers)
+
+    for repo in repos:
+        pull_requests = get_pull_requests(organization, repo['name'], headers)
+        for pr in pull_requests:
+            commits = get_commits_for_pull_request(organization, repo['name'], pr['number'], headers)
+            print(f'total number of commits in repo: {repo["name"]}, pr-#{pr["number"]}, pr-title: {pr["title"][:32]}, head: {pr["head"]["ref"][:16]} into base: {pr["base"]["ref"][:16]}  - {len(commits)}')
+            total_prs += 1
+            total_commits += len(commits)
+    
+    # print(f"Total number of commits in pull requests: {total_commits}")
+
+
+    annual_pr_events_count, annual_pr_count = total_commits, total_prs
```

Comment:
How are we determining that these are annual? I did see a reference to a year limit in some of the GitHub API docs, but I'm not clear it applies to the API endpoints used in this script. If the limit is pertinent, I recommend adding a comment where it applies that refers to the relevant doc.

Reply:
Good catch. I missed this one:
`commits_in_pr_repo_last_year = sum(1 for commit in commits if datetime.strptime(commit['commit']['author']['date'], '%Y-%m-%dT%H:%M:%SZ') > one_year_ago)`